### PR TITLE
Update to the latest rustc-serialize

### PR DESCRIPTION
**Pull request by bitonic**

### CODE REVIEW

File: src/test.rs [hunk 1]
```diff
@@ -14,18 +13,12 @@ use rustc_serialize::{
...
     decode,
-    SizeLimit
 };
 use super::SizeLimit::Infinite;
 
-fn the_same<'a,
-            V: Encodable<EncoderWriter<'a, MemWriter>, IoError>  +
-               Decodable<DecoderReader<'a, MemReader>, IoError> +
-               PartialEq + Show>(element: V) {
+fn the_same<'a, V>(element: V) where V: Encodable, V: Decodable, V: PartialEq, V: Show {
```
Comment (TyOverby) [inline src/test.rs]: Why isn't this `V: Encodable + Decodable + ` ...?

File: src/lib.rs [hunk 2]
```diff
@@ -43,19 +41,13 @@ where T: Decodable<DecoderReader<'a, MemReader>, IoError> {
 // the current json encoder in the stdlib
 
 // TODO: Make code safe https://github.com/rust-lang/rust/issues/14302
```
Comment (TyOverby) [inline src/lib.rs]: Remove comment.

File: src/lib.rs [hunk 3]
```diff
@@ -43,19 +41,13 @@ where T: Decodable<DecoderReader<'a, MemReader>, IoError> {
...
-pub fn encode_into<'a, W, T>(t: &T, w: &mut W, size_limit: SizeLimit) -> IoResult<()>
-where W: 'a + Writer, T: Encodable<EncoderWriter<'a, W>, IoError>{
-    unsafe {
-        t.encode(std::mem::transmute(&mut writer::EncoderWriter::new(w, size_limit)))
-    }
+pub fn encode_into<T: Encodable, W: Writer>(t: &T, w: &mut W, size_limit: SizeLimit) -> IoResult<()> {
+    t.encode(&mut writer::EncoderWriter::new(w, size_limit))
 }
 
 // TODO: Make code safe https://github.com/rust-lang/rust/issues/14302
```
Comment (TyOverby) [inline src/lib.rs]: Remove comment.

**Comment by TyOverby** [comment]
As an aside, feel free to add yourself to the owners list in `Cargo.toml`.  I've also added you as an owner of the github repo.

File: src/test.rs (hunk 1 above)
Comment (bitonic) [inline src/test.rs]: I find the `where` notation nicer for long lists, and I believe the `+` syntax is sort of frowned upon.

In other words, no good reason :).

File: src/lib.rs (hunk 2 above)
Comment (bitonic) [inline src/lib.rs]: Ooops, will do.

**Pull request merged**